vendor/bat-native-ledger/src/bat/ledger/internal/state/current_reconcile_state.cc

```diff
@@ -0,0 +1,381 @@
+      const std::string currency_code = rate.first;
+
+      const auto& exchange_rate_value = rate.second;
+      const double exchange_rate = exchange_rate_value.GetDouble();
+
+      current_reconcile.rates.insert({currency_code, exchange_rate});
+    }
+  }
+
+  // Reconcile Directions
+  const auto* reconcile_directions_list =
+      dictionary->FindListKey(kDirectionsKey);
+  if (reconcile_directions_list) {
+    const ReconcileDirectionState reconcile_direction_state;
+    for (const auto& reconcile_direction_value :
+        reconcile_directions_list->GetList()) {
+      if (!reconcile_direction_value.is_dict()) {
+        NOTREACHED();
+        continue;
+      }
+
+      const base::DictionaryValue* reconcile_direction_dictionary = nullptr;
+      reconcile_direction_value.GetAsDictionary(
+          &reconcile_direction_dictionary);
+      if (!reconcile_direction_dictionary) {
+        NOTREACHED();
+        continue;
+      }
+
+      ReconcileDirectionProperties reconcile_direction;
+      if (!reconcile_direction_state.FromDict(reconcile_direction_dictionary,
+          &reconcile_direction)) {
+        continue;
+      }
+
+      current_reconcile.reconcile_directions.push_back(reconcile_direction);
+    }
+  }
+
+  // Transition legacy list to reconcile directions
+  const auto* list_list = dictionary->FindListKey(kListKey);
+  if (list_list) {
+    for (const auto& list_value : list_list->GetList()) {
+      if (!list_value.is_dict()) {
+        NOTREACHED();
+        continue;
+      }
+
+      const base::DictionaryValue* list_dictionary = nullptr;
+      list_value.GetAsDictionary(&list_dictionary);
+      if (!list_dictionary) {
+        NOTREACHED();
+        continue;
+      }
+
+      ReconcileDirectionProperties reconcile_direction;
+
+      const auto* id = list_dictionary->FindStringKey(kIdKey);
+      if (!id) {
+        NOTREACHED();
+        continue;
+      }
+      reconcile_direction.publisher_key = *id;
+
+      const auto weight = list_dictionary->FindDoubleKey(kWeightKey);
+      if (!weight) {
+        NOTREACHED();
+        continue;
+      }
+      reconcile_direction.amount_percent = *weight;
+
+      current_reconcile.reconcile_directions.push_back(reconcile_direction);
+    }
+  }
+
+  // Retry Step
+  const auto retry_step = dictionary->FindIntKey(kRetryStepKey);
+  if (retry_step) {
+    current_reconcile.retry_step = static_cast<ContributionRetry>(*retry_step);
+  }
+
+  // Retry Level
+  const auto retry_level = dictionary->FindIntKey(kRetryLevelKey);
+  if (retry_level) {
+    current_reconcile.retry_level = *retry_level;
+  }
+
+  // Destination
+  const auto* destination = dictionary->FindStringKey(kDestinationKey);
+  if (destination) {
+    current_reconcile.destination = *destination;
+  }
+
+  // Proof
+  const auto* proof = dictionary->FindStringKey(kProofKey);
+  if (proof) {
+    current_reconcile.proof = *proof;
+  }
+
+  *properties = current_reconcile;
+
+  return true;
```

Comment:
Would recommend a ticket to refactor into smaller methods

Reply:
In fact, this applies across the whole PR - and as I understand it, this is not 'new' code but existing relocated code - so I think a general 'fix up' refactor ticket is in order.

Reply:
FromJson code is new however as only for migration did not decouple